src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,35 @@
+import { MouseEvent } from 'react';
+import { Page } from '../../services/api';
+import styles from './Pagination.module.css';
+
+interface PaginationProps {
+  page: Page;
+  setActivePage: (pageNumber: string | null) => void;
+}
+
+export function Pagination(props: PaginationProps) {
+  const paginationNumbers = [];
+  const pagesInfo = props.page;
+
+  const clickPageButtonsHandler = (e: MouseEvent) => {
+    if (e.target instanceof HTMLButtonElement) {
+      props.setActivePage(e.target.textContent);
+    }
+  };
+
+  for (let i = 1; i < pagesInfo.totalPages; i++) {
```

Comment:
you will run this every time you update any page. to get rid of this you can use memorization, for example
```typescript
export function Pagination({ page, setActivePage }: PaginationProps) {
  const pager = useMemo(
    () => new Array(page.totalPages).fill(null),
    [page.totalPages],
  );

  const clickPageButtonsHandler = (e: MouseEvent) => {
    if (e.target instanceof HTMLButtonElement) {
      setActivePage(e.target.textContent);
    }
  };

  return (
    <div className={styles.pagination} onClick={clickPageButtonsHandler}>
      {pager.map((_, index) => (
        <button
          key={index}
          className={`${page.pageNumber === index + 1 ? styles.active : ''}`}
        >
          {index + 1}
        </button>
      ))}
    </div>
  );
}
```

Reply:
Fixed.